pinot-common/src/main/java/org/apache/pinot/common/utils/config/InstanceUtils.java

```diff
@@ -119,4 +122,36 @@ public static void updateHelixInstanceConfig(InstanceConfig instanceConfig, Inst
       mapFields.remove(POOL_KEY);
     }
   }
+
+  /**
+   * Returns an instance admin URL from a given InstanceConfig. Will decipher whether the URL protocol should
+   * be HTTP vs HTTPS.
+   */
+  public static URL getInstanceURL(InstanceConfig instanceConfig) {
+    String hostname = instanceConfig.getHostName();
+    if (hostname.startsWith(Helix.PREFIX_OF_SERVER_INSTANCE)) {
```

Comment:
hmm, so this is for server prefix?

Reply:
Will create a method that supports all the components, checking for the different prefixes.